src/lib.rs

```diff
@@ -258,9 +260,9 @@ impl System {
     /// created on its own thread instead of the calling thread.
     /// Returns a [`SpawnBuilder`] which can be used to customize the
     /// spawning of the actor.
-    pub fn prepare_fn<A, F>(&mut self, factory: F) -> SpawnBuilder<A>
+    pub fn prepare_fn<A, F>(&mut self, factory: F) -> SpawnBuilder<A, impl FnOnce() -> A>
```

Comment:
This is more explicit, I think.
```suggestion
    pub fn prepare_fn<A, F>(&mut self, factory: F) -> SpawnBuilder<A, F>
```

Reply:
Oh yeah totally.